src/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -465,26 +465,72 @@ class OSRow extends React.Component {
     }
 }
 
-const OfflineTokenRow = ({ offlineToken, onValueChanged, validationFailed }) => {
-    const validationStateToken = validationFailed.offlineToken ? 'error' : 'default';
+const OfflineTokenRow = ({ onValueChanged, validationFailed }) => {
+    const link = <a href="https://access.redhat.com/management/api" target="_blank" rel="noopener noreferrer">
+        <ExternalLinkAltIcon className="pf-u-mr-xs" />
+        {_("Get a new RHSM token.")}
+    </a>;
+    const validationStates = {
+        DEFAULT: {
+            option: validationFailed.accessToken ? "error" : "default",
+            message: validationFailed.accessToken || <span>
+                { link }
+                {" " + _(" Then copy the token and paste it above.")}
+            </span>,
+        },
+        INPROGRESS: {
+            option: "default",
+            message: _("Validating..."),
+        },
+        FAILED: {
+            option: "error",
+            message: <span className="invalid-token-helper">
+                {_("Invalid offline token.") + " "}
+                { link }
+            </span>,
+        },
+        SUCCESS: {
+            option: "success",
+            message: _("Valid token."),
+        },
+    };
+
+    const [validationState, setValidationState] = useState(validationStates.DEFAULT);
+    const [offlineToken, setOfflineToken] = useState("");
+
+    const setOfflineTokenHelper = (offlineToken) => {
+        setOfflineToken(offlineToken);
+
+        if (isEmpty(offlineToken)) {
+            setValidationState(validationStates.DEFAULT);
+            onValueChanged("accessToken", "");
+        } else {
+            setValidationState(validationStates.INPROGRESS);
+            getAccessToken(offlineToken)
+                    .then(out => {
+                        const accessToken = out.trim();
+                        onValueChanged("accessToken", accessToken);
+                        setValidationState(validationStates.SUCCESS);
+                    })
+                    .catch(ex => {
+                        console.error(`Offline token validation failed: "${JSON.stringify(ex)}"`);
+                        onValueChanged("accessToken", "");
+                        setValidationState(validationStates.FAILED);
```

Comment:
Can you please check the "success" and "failure" code paths in tests? I suggest to split the PR to do the "check token" part first, as that otherwise looks fine.